### PR TITLE
1340 detectors in tobyfit

### PR DESCRIPTION
**Pull request by tgperring**
Completes task of ticket #1340 which requires additional work to allow sqw objects with multiple IX_detector_arrays. The major changes are:

(1) The code inside tobyfit_DGfermi_resconv_init (and tobyfit_DGdisk_resconv_init) which fills an object_lookup object with the detector information has been isolated in one function (detector_array) and a call to object_lookup. This code will need to be replaced once the IX_detector_array completely replaces the old detpar (work being performed in parallel by Chris Marooney, ticket #1398)

(2) The extended capabilities of object_lookup/func_eval_ind use the detector object_lookup created above to index both irun and idet in the calculation of dq_mat and qw (the unsymmetrised values of h,k,l,e for each pixel)

(3) dq_mat computation has been delegated to tobyfit_DGfermi_resoconv and tobyfit_DGfermi_resoconv to be performed on the fly. This prepares Tobyfit for chunking/parallelism to reduce memory use by Tobyfit [ticket still to be written?]

(4) Changes to tobyfit_DGfermi[disk]_resfun_covariance to accommodate the above changes

Also includes updated mexw64 files. This supesedes PR 1396. Added inadvertently, but I think is identical except for being built on Toby Perring's Win laptop instead of Alex Buts'.

### CODE REVIEW

**Review by cmarooney-stfc — APPROVED** [review]

File: horace_core/sqw/@sqw/tobyfit_DGdisk_resconv.m [hunk 1]
```diff
@@ -185,21 +185,22 @@
...
-    detdcn=lookup.detdcn{iw};
-    x2=lookup.x2{iw};
     dt=lookup.dt{iw};
     qw=lookup.qw{iw};
-    dq_mat=lookup.dq_mat{iw};
 
     % Run and detector for each pixel
-    run_idx = win(i).pix.run_idx';   % column vector
+    irun = win(i).pix.run_idx';   % column vector
+    idet = win(i).pix.detector_idx';   % column vector
```
[the remark this answers is not in the document]
Comment (tgperring) [reply horace_core/sqw/@sqw/tobyfit_DGdisk_resconv.m:197]: This was existing code - I just cut-n-paste idet and npix up from a position 20 lines further down in the code in order to collect them together, just sa they had already been done in tobyfit_DGfermi_resconv. Happy to change.

File: horace_core/Tobyfit/detector_array.m [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    % no detector info was available when the sqw was populated, so
+    % continue with the old detector initialisation from detpar
+    if use_tubes
+        detectors = IX_detector_array (detpar.group, detpar.x2(:), ...
+            detpar.phi(:), detpar.azim(:),...
+            IX_det_He3tube (detpar.width, detpar.height, 6.35e-4, 10));   % 10atms, wall thickness=0.635mm
+    else
+        detectors = IX_detector_array (detpar.group, detpar.x2(:), detpar.phi(:), detpar.azim(:),...
+            IX_det_TobyfitClassic (detpar.width, detpar.height));
+    end
```
Comment (abuts) [inline horace_core/Tobyfit/detector_array.m:43]: I though we deprecate detpar and use `experiment_info.detector_arrays` everywhere. 

The code, presumably correctly, converts `detpar` to `det_array` but why is it necessary anywhere except the conversion of old file formats?

**Review by abuts — APPROVED** [review]
The changes seems overlap with @cmarooney-stfc work on using detector arrays everywhere within the code and should be reconciled with them. 

I would overload `sqw.detpar` to return detector arrays but its difficult to judge joint wor. 

As common comment to common practice -- every piece of code needs its unit test. 

mex here -- ok, I will cancele my PR in favour of this one.

File: horace_core/Tobyfit/detector_array.m (hunk 2 above)
Comment (cmarooney-stfc) [reply horace_core/Tobyfit/detector_array.m:43]: @abuts Alex, you're correct that eliminating detpar is the ultimate goal, but it's still a work in progress. This PR gets us further towards it

**Comment by tgperring** [comment]
Merging as the changes in response to the review were tiny and highly localised , and passed all coupled tests in Horace on my laptop.

**Pull request merged**